azure/bastion/locals.tf

```diff
@@ -0,0 +1,3 @@
+locals {
+  whitelist = try(length(var.whitelist) == 0, false) ? ["Internet"] : var.whitelist
```

Comment:
Changed to length(var.whitelist) >= 1 ? var.whitelist : ["Internet"]

Reply:
@jk-ans - What are your thoughts on removing this local as per that comment thread?

Reply:
To quote:

```
If the end user does pass in an empty list and knowingly overwrites default value, then should it be the modules responsibility to correct the user for their behaviour without the correction being highlighted? Could we not just use validation on the variable instead?

In the case of the module within the module, the parent module should have the check to ensure that the empty list cannot be passed in, I do not believe that this should be the responsibility of the child module to make the correction on behalf of the user as it may lead to frustration where the module is doing unexpected behaviour.
```

Reply:
So what I mean by this is:

- Set the default value of the variable to `["Internet"]`
- Add variable validation to the variable to prevent empty lists from being passed into the module.

Reply:
removed locals file  and replaced local.whitelist with var.whitelist in code